### DISCUSSION TITLE
[Question][Hw3] Reason for the partial credit.

### DISCUSSION THREAD

**Discussion by JaejunHan** [prosyslab-classroom/cs348-information-security#697] [Q&A]
Name: Jaejun Han

In HW3, Gradescope indicates that all testcases are passed. (keygen, enc, dec, and crack)
However, I got minor minus in somewhere.
Could I know the reason for this? Thank you.

**Comment by leejy12** [comment]
If all the test cases pass but your score isn't 100%, then you probably have some room for improving the running time.

https://github.com/prosyslab-classroom/cs348-information-security/issues/270#issue-1663691701:
> Memory and time limits are important in this assignment.
> For each test case, score will be given as follow:

> ```
> let score = match (execute test_case) with
>  | Success(execution_time) -> 60.0 - execution_time
>  | Fail -> 0.0
>  | Timeout -> 0.0
> ```

**Comment by yeonhee-ryou** [comment]
Please check the **Auto grading** section #270
